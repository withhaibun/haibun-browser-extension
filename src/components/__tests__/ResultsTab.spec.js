import { mount } from '@vue/test-utils'
import VueHighlightJS from 'vue3-highlightjs'

import ResultsTab from '../ResultsTab'

describe('RecordingTab.vue', () => {
  test('it has the correct pristine / empty state', () => {
    const wrapper = mount(ResultsTab)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('code.javascript').exists()).toBe(false)
  })

  test('it show a code box when there is code', () => {
    const wrapper = mount(ResultsTab, {
      global: {
        plugins: [VueHighlightJS],
      },
      props: { puppeteer: `await page.click('.class')` },
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('code.javascript').exists()).toBe(true)
  })

  test('it render tabs for haibun', () => {
    const wrapper = mount(ResultsTab)
    expect(wrapper.findAll('.tabs__action').length).toEqual(1)
  })

})
