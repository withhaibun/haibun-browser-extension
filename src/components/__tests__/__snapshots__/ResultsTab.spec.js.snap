// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordingTab.vue it has the correct pristine / empty state 1`] = `
<div
  class="tab results-tab"
>
  <div
    class="tabs"
  >
    
    <button
      class="tabs__action selected"
    >
      <!--v-if-->
      <img
        src=""
        width="16"
      />
      <span
        class="tabs__action--text"
      >
        puppeteer
      </span>
    </button>
    <button
      class="tabs__action"
    >
      <span>
        ðŸŽ­
      </span>
      <!--v-if-->
      <span
        class="tabs__action--text"
      >
        playwright
      </span>
    </button>
    
  </div>
  <div
    class="content"
  >
    <pre>
              
      <code>
        No code yet...
      </code>
      
      
    </pre>
  </div>
</div>
`;

exports[`RecordingTab.vue it show a code box when there is code 1`] = `
<div
  class="tab results-tab"
>
  <div
    class="tabs"
  >
    
    <button
      class="tabs__action selected"
    >
      <!--v-if-->
      <img
        src=""
        width="16"
      />
      <span
        class="tabs__action--text"
      >
        puppeteer
      </span>
    </button>
    <button
      class="tabs__action"
    >
      <span>
        ðŸŽ­
      </span>
      <!--v-if-->
      <span
        class="tabs__action--text"
      >
        playwright
      </span>
    </button>
    
  </div>
  <div
    class="content"
  >
    <pre>
              
      <code
        class="javascript hljs"
      >
        <span
          class="hljs-keyword"
        >
          await
        </span>
         page.click(
        <span
          class="hljs-string"
        >
          '.class'
        </span>
        )
      </code>
      
      
    </pre>
  </div>
</div>
`;
